presto-main/src/main/java/io/prestosql/security/AccessControl.java

```diff
@@ -255,6 +255,13 @@
      */
     void checkCanCreateViewWithSelectFromColumns(SecurityContext context, QualifiedObjectName tableName, Set<String> columnNames);
 
+    /**
+     * Check if identity is allowed to create a view that executes the function.
+     *
+     * @throws io.prestosql.spi.security.AccessDeniedException if not allowed
+     */
+    void checkCanCreateViewWithExecuteFunction(SecurityContext context, String functionName);
```

Comment:
I think this is the wrong abstraction. There's nothing in the SQL spec that limits whether user can create a view based on whether it contains certain functions. All it says is that:

```
A column privilege descriptor is created that defines the privilege SELECT on CV to A. 
That privilege is grantable if and only if all the following are true:
...
2) The applicable privileges for A include grantable EXECUTE privileges on all SQL-invoked 
routines that are subject routines of <routine invocation>s contained in QE.
...
```

which, basically, states that the creator of the view (i.e., the schema owner) has a SELECT privilege on each column of the view. That privilege is grantable to others if certain conditions apply (i.e., among other things, the creator has a grantable EXECUTE privilege on all functions referenced by the query)